x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsManager.java

```diff
@@ -90,6 +93,25 @@ public void execute(DataFrameAnalyticsTask task, DataFrameAnalyticsState current
         // With config in hand, determine action to take
         ActionListener<DataFrameAnalyticsConfig> configListener = ActionListener.wrap(
             config -> {
+                // Check if existing destination index is incompatible.
+                // If it is, we delete it and start from reindexing.
+                IndexMetadata destIndex = clusterState.getMetadata().index(config.getDest().getIndex());
+                if (destIndex != null) {
+                    MappingMetadata destIndexMapping = clusterState.getMetadata().index(config.getDest().getIndex()).mapping();
+                    if (DestinationIndex.isCompatible(config.getId(), destIndexMapping) == false) {
+                        LOGGER.info("[{}] Destination index is out of date; will delete and reindex from scratch", config.getId());
```

Comment:
In my opinion (loosely held) it would be good to say "index was created in version X, but minimum supported version is version Y. Starting from scratch" or something.

Reply:
Yeah, I agree. It just makes the code a bit uglier as I have to get the version separately. But it's probably worth it. I'll do that which means I can remove entirely the other debug message about the version.